projects/kubernetes/kubernetes/1-21/patches/0021-EKS-PATCH-Parse-ipv6-address-before-comparison-10773.patch

```diff
@@ -0,0 +1,57 @@
+From 9386bf0ea183c589f53e8a0c6e4cf58dd3920f69 Mon Sep 17 00:00:00 2001
+From: Jyoti Mahapatra <jyotima@amazon.com>
+Date: Sat, 22 Jan 2022 01:16:11 +0000
+Subject: --EKS-PATCH-- Parse ipv6 address before comparison (#107736)
+
+Cherry-pick of upstream Kubernetes:
+https://github.com/kubernetes/kubernetes/pull/107736, which is
+included in Kubernetes 1.24.
+
+This fixes a bug related to ipv6 clusters, as described in Kubernetes
+issue #107735 (https://github.com/kubernetes/kubernetes/issues/107735)
+
+Signed-off-by: Jyoti Mahapatra <jyotima@amazon.com>
+---
+ pkg/kubelet/nodestatus/setters.go      |  6 ++++--
+ pkg/kubelet/nodestatus/setters_test.go | 14 ++++++++++++++
+ 2 files changed, 15 insertions(+), 1 deletion(-)
+
+diff --git a/pkg/kubelet/nodestatus/setters.go b/pkg/kubelet/nodestatus/setters.go
+index 52f12178b66..a31cfbd2970 100644
+--- a/pkg/kubelet/nodestatus/setters.go
++++ b/pkg/kubelet/nodestatus/setters.go
+@@ -125,8 +125,10 @@ func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
+ 
+ 				nodeIPTypes := make(map[v1.NodeAddressType]bool)
+ 				for _, nodeAddress := range cloudNodeAddresses {
+-					if nodeAddress.Address == nodeIP.String() {
++					if netutils.ParseIPSloppy(nodeAddress.Address).Equal(nodeIP) {
```

Comment:
Does parseipsloppy exists yet in 1-21/1-22?  Im not really sure, but I thought that was added after they bumped to golang 1.17 which changed the parsing rules.  The BR patch does it slightly differently, https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/kubernetes-1.21/0002-IPv6-node-address-before-comparing-with-node-I.patch#L29